src/api.js

```diff
@@ -72,10 +72,22 @@ const parseActor = rawActor => {
   };
```

Comment:
// TODO: currentDate field,
before we compute the remaining Time based on instancing Date() for each row,  this is wrong since the progress can increment during these process of creation, "we can get a considerable difference if len(messages) is large enough" to produce an accurate result of remaining time and progress, now we have independent current Dates for each component, which is also not precise, the ideal will be having a global current date that will make possible removing the dependency of time mock, however, the coupling could increment.